File: Axiom/Internal/Ast/IfStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Axiom.Internal.Ast
{
    internal class IfStatement : Statement
    {
        public IfStatement(Expression condition, Statement body, Statement elseBody = null)
        {
            Condition = condition;
            Body = body;
            ElseBody = elseBody;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public Expression Condition { get; private set; }
        public Statement Body { get; set; }
        public Statement ElseBody { get; set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/IfStatement.java
package com.moybl.axiom.ast;

import com.moybl.axiom.Position;

public class IfStatement extends Statement {

	private Expression condition;
	private Statement body;
	private Statement elseBody;

	public IfStatement(Position position, Expression condition, Statement body, Statement elseBody) {
		super(position);
		this.condition = condition;
		this.body = body;
		this.elseBody = elseBody;
	}

	public IfStatement(Position position, Expression condition, Statement body) {
		super(position);
		this.condition = condition;
		this.body = body;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Expression getCondition() {
		return condition;
	}

	public Statement getBody() {
		return body;
	}

	public Statement getElseBody() {
		return elseBody;
	}

}
